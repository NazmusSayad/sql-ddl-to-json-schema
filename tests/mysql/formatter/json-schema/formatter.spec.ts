import { getJSONSchemaFormat } from '../../parse-handler';
import { run } from '../../runner';
import createTable from './sql/create-table';

const sql = [createTable];

run((query) => getJSONSchemaFormat(query, { useRef: true }), {
  'JSON Schema formatter: should format tables correctly, with ref.': {
    queries: [sql.join('')],
  },
});

run((query) => getJSONSchemaFormat(query, { useRef: false }), {
  'JSON Schema formatter: should format tables correctly, without ref.': {
    queries: [sql.join('')],
  },
});
