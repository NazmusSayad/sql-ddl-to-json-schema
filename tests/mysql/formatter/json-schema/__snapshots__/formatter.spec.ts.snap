// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Schema formatter: should format tables correctly, with ref. 1`] = `
[
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for person table",
    "$id": "person",
    "title": "person",
    "description": "test table",
    "type": "object",
    "required": [
      "id",
      "name",
      "binarynickname"
    ],
    "definitions": {
      "id": {
        "$comment": "primary key",
        "description": "primary key test",
        "type": "integer",
        "minimum": 1,
        "maximum": 4294967295
      },
      "name": {
        "type": "string",
        "maxLength": 65535,
        "default": "Jane \\"Doe\\" O\`neil"
      },
      "nickname": {
        "type": "string",
        "maxLength": 20,
        "default": "J"
      },
      "binarynickname": {
        "type": "string",
        "maxLength": 20
      },
      "ssn": {
        "description": "SMALLINT",
        "type": "integer",
        "minimum": -32768,
        "maximum": 32767
      },
      "ssn1": {
        "description": "SMALLINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 65535
      },
      "ssn2": {
        "description": "TINYINT",
        "type": "integer",
        "minimum": -128,
        "maximum": 127
      },
      "ssn3": {
        "description": "TINYINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 255
      },
      "ssn4": {
        "description": "INT",
        "type": "integer",
        "minimum": -2147483648,
        "maximum": 2147483647
      },
      "ssn5": {
        "description": "INT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 4294967295
      },
      "ssn6": {
        "description": "MEDIUMINT",
        "type": "integer",
        "minimum": -8388608,
        "maximum": 8388607
      },
      "ssn7": {
        "description": "MEDIUMINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 16777215
      },
      "ssn8": {
        "description": "BIGINT",
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "-9223372036854775808",
        "x-bigint-maximum": "9223372036854775807"
      },
      "ssn9": {
        "description": "BIGINT UNSIGNED",
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "0",
        "x-bigint-maximum": "18446744073709551615"
      },
      "height": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": -9999999999,
        "default": null
      },
      "weight": {
        "type": "number",
        "maximum": 999,
        "minimum": -999
      },
      "status": {
        "type": "string"
      },
      "dob": {
        "type": "string",
        "format": "date",
        "default": "NOW()"
      },
      "tob": {
        "type": "string",
        "format": "time"
      },
      "initials": {
        "type": "string",
        "maxLength": 3
      },
      "prefix": {
        "type": "string",
        "maxLength": 3
      },
      "suffix": {
        "type": "string",
        "maxLength": 1
      },
      "nsuffix": {
        "type": "string",
        "maxLength": 3
      },
      "sequence": {
        "type": "string",
        "maxLength": 20
      },
      "avatar": {
        "type": "string"
      },
      "motto": {
        "type": "string",
        "maxLength": 50
      },
      "history": {
        "type": "string",
        "maxLength": 255
      },
      "gender": {
        "type": "string",
        "enum": [
          "M",
          "F"
        ]
      },
      "shape": {
        "type": "string"
      },
      "pattern": {
        "type": "string"
      }
    },
    "properties": {
      "id": {
        "$ref": "#/definitions/id"
      },
      "name": {
        "$ref": "#/definitions/name"
      },
      "nickname": {
        "$ref": "#/definitions/nickname"
      },
      "binarynickname": {
        "$ref": "#/definitions/binarynickname"
      },
      "ssn": {
        "$ref": "#/definitions/ssn"
      },
      "ssn1": {
        "$ref": "#/definitions/ssn1"
      },
      "ssn2": {
        "$ref": "#/definitions/ssn2"
      },
      "ssn3": {
        "$ref": "#/definitions/ssn3"
      },
      "ssn4": {
        "$ref": "#/definitions/ssn4"
      },
      "ssn5": {
        "$ref": "#/definitions/ssn5"
      },
      "ssn6": {
        "$ref": "#/definitions/ssn6"
      },
      "ssn7": {
        "$ref": "#/definitions/ssn7"
      },
      "ssn8": {
        "$ref": "#/definitions/ssn8"
      },
      "ssn9": {
        "$ref": "#/definitions/ssn9"
      },
      "height": {
        "$ref": "#/definitions/height"
      },
      "weight": {
        "$ref": "#/definitions/weight"
      },
      "status": {
        "$ref": "#/definitions/status"
      },
      "dob": {
        "$ref": "#/definitions/dob"
      },
      "tob": {
        "$ref": "#/definitions/tob"
      },
      "initials": {
        "$ref": "#/definitions/initials"
      },
      "prefix": {
        "$ref": "#/definitions/prefix"
      },
      "suffix": {
        "$ref": "#/definitions/suffix"
      },
      "nsuffix": {
        "$ref": "#/definitions/nsuffix"
      },
      "sequence": {
        "$ref": "#/definitions/sequence"
      },
      "avatar": {
        "$ref": "#/definitions/avatar"
      },
      "motto": {
        "$ref": "#/definitions/motto"
      },
      "history": {
        "$ref": "#/definitions/history"
      },
      "gender": {
        "$ref": "#/definitions/gender"
      },
      "shape": {
        "$ref": "#/definitions/shape"
      },
      "pattern": {
        "$ref": "#/definitions/pattern"
      }
    }
  },
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for pet table",
    "$id": "pet",
    "title": "pet",
    "type": "object",
    "required": [
      "unique_id1"
    ],
    "definitions": {
      "id": {
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "-9223372036854775808",
        "x-bigint-maximum": "9223372036854775807"
      },
      "species": {
        "type": "string",
        "maxLength": 200
      },
      "owner_id": {
        "type": "integer",
        "minimum": -8388608,
        "maximum": 8388607
      },
      "height_unsigned": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": 0
      },
      "weight_unsigned": {
        "type": "number",
        "maximum": 99.99,
        "minimum": 0
      },
      "height": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": -9999999999
      },
      "weight": {
        "type": "number",
        "maximum": 99.99,
        "minimum": -99.99
      },
      "status": {
        "type": "string"
      },
      "birth": {
        "type": "string",
        "format": "date-time"
      },
      "year": {
        "type": "string",
        "pattern": "\\\\d{1,2}"
      },
      "initial": {
        "type": "string",
        "maxLength": 1
      },
      "avatar": {
        "type": "string"
      },
      "photo": {
        "type": "string"
      },
      "history": {
        "type": "string",
        "maxLength": 16777215
      },
      "gender": {
        "type": "string",
        "pattern": "^(M|F)(,(M|F))*$"
      },
      "intelligent": {
        "type": "string",
        "pattern": "^(N)(,(N))*$"
      },
      "shape": {
        "type": "string"
      },
      "pattern": {
        "type": "string"
      },
      "unique_id1": {
        "$comment": "primary key",
        "type": "string",
        "pattern": "^[a-f\\\\d]{8}-([a-f\\\\d]{4}-){3}[a-f\\\\d]{12}$"
      },
      "unique_id2": {
        "type": "string",
        "pattern": "^[a-f\\\\d]{8}-([a-f\\\\d]{4}-){3}[a-f\\\\d]{12}$"
      },
      "object": {
        "type": "string"
      }
    },
    "properties": {
      "id": {
        "$ref": "#/definitions/id"
      },
      "species": {
        "$ref": "#/definitions/species"
      },
      "owner_id": {
        "$ref": "#/definitions/owner_id"
      },
      "height_unsigned": {
        "$ref": "#/definitions/height_unsigned"
      },
      "weight_unsigned": {
        "$ref": "#/definitions/weight_unsigned"
      },
      "height": {
        "$ref": "#/definitions/height"
      },
      "weight": {
        "$ref": "#/definitions/weight"
      },
      "status": {
        "$ref": "#/definitions/status"
      },
      "birth": {
        "$ref": "#/definitions/birth"
      },
      "year": {
        "$ref": "#/definitions/year"
      },
      "initial": {
        "$ref": "#/definitions/initial"
      },
      "avatar": {
        "$ref": "#/definitions/avatar"
      },
      "photo": {
        "$ref": "#/definitions/photo"
      },
      "history": {
        "$ref": "#/definitions/history"
      },
      "gender": {
        "$ref": "#/definitions/gender"
      },
      "intelligent": {
        "$ref": "#/definitions/intelligent"
      },
      "shape": {
        "$ref": "#/definitions/shape"
      },
      "pattern": {
        "$ref": "#/definitions/pattern"
      },
      "unique_id1": {
        "$ref": "#/definitions/unique_id1"
      },
      "unique_id2": {
        "$ref": "#/definitions/unique_id2"
      },
      "object": {
        "$ref": "#/definitions/object"
      }
    }
  },
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for house table",
    "$id": "house",
    "title": "house",
    "type": "object",
    "required": [
      "id"
    ],
    "definitions": {
      "id": {
        "$comment": "primary key",
        "type": "integer",
        "minimum": 1,
        "maximum": 2147483647
      },
      "pet_id": {
        "type": "integer",
        "minimum": -128,
        "maximum": 127
      },
      "coordx": {
        "type": "number",
        "maximum": 9999.99,
        "minimum": -9999.99
      },
      "coordy": {
        "type": "number"
      },
      "letter": {
        "type": "string",
        "maxLength": 1
      },
      "is_built": {
        "type": "boolean"
      },
      "is_apartment": {
        "type": "boolean"
      },
      "updated_at": {
        "type": "string"
      },
      "year": {
        "type": "string",
        "pattern": "\\\\d{1,4}"
      },
      "sequence": {
        "type": "string",
        "maxLength": 1,
        "default": "b'00101'"
      },
      "photo": {
        "type": "string"
      },
      "history": {
        "type": "string",
        "maxLength": 4294967295
      },
      "gender": {
        "type": "string",
        "enum": [
          "X"
        ]
      },
      "size": {
        "type": "string"
      },
      "where": {
        "type": "string"
      },
      "street": {
        "type": "string"
      },
      "neighbors": {
        "type": "string"
      },
      "city": {
        "type": "string"
      }
    },
    "properties": {
      "id": {
        "$ref": "#/definitions/id"
      },
      "pet_id": {
        "$ref": "#/definitions/pet_id"
      },
      "coordx": {
        "$ref": "#/definitions/coordx"
      },
      "coordy": {
        "$ref": "#/definitions/coordy"
      },
      "letter": {
        "$ref": "#/definitions/letter"
      },
      "is_built": {
        "$ref": "#/definitions/is_built"
      },
      "is_apartment": {
        "$ref": "#/definitions/is_apartment"
      },
      "updated_at": {
        "$ref": "#/definitions/updated_at"
      },
      "year": {
        "$ref": "#/definitions/year"
      },
      "sequence": {
        "$ref": "#/definitions/sequence"
      },
      "photo": {
        "$ref": "#/definitions/photo"
      },
      "history": {
        "$ref": "#/definitions/history"
      },
      "gender": {
        "$ref": "#/definitions/gender"
      },
      "size": {
        "$ref": "#/definitions/size"
      },
      "where": {
        "$ref": "#/definitions/where"
      },
      "street": {
        "$ref": "#/definitions/street"
      },
      "neighbors": {
        "$ref": "#/definitions/neighbors"
      },
      "city": {
        "$ref": "#/definitions/city"
      }
    }
  }
]
`;

exports[`JSON Schema formatter: should format tables correctly, without ref. 1`] = `
[
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for person table",
    "$id": "person",
    "title": "person",
    "description": "test table",
    "type": "object",
    "required": [
      "id",
      "name",
      "binarynickname"
    ],
    "properties": {
      "id": {
        "$comment": "primary key",
        "description": "primary key test",
        "type": "integer",
        "minimum": 1,
        "maximum": 4294967295
      },
      "name": {
        "type": "string",
        "maxLength": 65535,
        "default": "Jane \\"Doe\\" O\`neil"
      },
      "nickname": {
        "type": "string",
        "maxLength": 20,
        "default": "J"
      },
      "binarynickname": {
        "type": "string",
        "maxLength": 20
      },
      "ssn": {
        "description": "SMALLINT",
        "type": "integer",
        "minimum": -32768,
        "maximum": 32767
      },
      "ssn1": {
        "description": "SMALLINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 65535
      },
      "ssn2": {
        "description": "TINYINT",
        "type": "integer",
        "minimum": -128,
        "maximum": 127
      },
      "ssn3": {
        "description": "TINYINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 255
      },
      "ssn4": {
        "description": "INT",
        "type": "integer",
        "minimum": -2147483648,
        "maximum": 2147483647
      },
      "ssn5": {
        "description": "INT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 4294967295
      },
      "ssn6": {
        "description": "MEDIUMINT",
        "type": "integer",
        "minimum": -8388608,
        "maximum": 8388607
      },
      "ssn7": {
        "description": "MEDIUMINT UNSIGNED",
        "type": "integer",
        "minimum": 0,
        "maximum": 16777215
      },
      "ssn8": {
        "description": "BIGINT",
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "-9223372036854775808",
        "x-bigint-maximum": "9223372036854775807"
      },
      "ssn9": {
        "description": "BIGINT UNSIGNED",
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "0",
        "x-bigint-maximum": "18446744073709551615"
      },
      "height": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": -9999999999,
        "default": null
      },
      "weight": {
        "type": "number",
        "maximum": 999,
        "minimum": -999
      },
      "status": {
        "type": "string"
      },
      "dob": {
        "type": "string",
        "format": "date",
        "default": "NOW()"
      },
      "tob": {
        "type": "string",
        "format": "time"
      },
      "initials": {
        "type": "string",
        "maxLength": 3
      },
      "prefix": {
        "type": "string",
        "maxLength": 3
      },
      "suffix": {
        "type": "string",
        "maxLength": 1
      },
      "nsuffix": {
        "type": "string",
        "maxLength": 3
      },
      "sequence": {
        "type": "string",
        "maxLength": 20
      },
      "avatar": {
        "type": "string"
      },
      "motto": {
        "type": "string",
        "maxLength": 50
      },
      "history": {
        "type": "string",
        "maxLength": 255
      },
      "gender": {
        "type": "string",
        "enum": [
          "M",
          "F"
        ]
      },
      "shape": {
        "type": "string"
      },
      "pattern": {
        "type": "string"
      }
    }
  },
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for pet table",
    "$id": "pet",
    "title": "pet",
    "type": "object",
    "required": [
      "unique_id1"
    ],
    "properties": {
      "id": {
        "type": "integer",
        "x-bigint": true,
        "x-bigint-minimum": "-9223372036854775808",
        "x-bigint-maximum": "9223372036854775807"
      },
      "species": {
        "type": "string",
        "maxLength": 200
      },
      "owner_id": {
        "type": "integer",
        "minimum": -8388608,
        "maximum": 8388607
      },
      "height_unsigned": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": 0
      },
      "weight_unsigned": {
        "type": "number",
        "maximum": 99.99,
        "minimum": 0
      },
      "height": {
        "type": "number",
        "maximum": 9999999999,
        "minimum": -9999999999
      },
      "weight": {
        "type": "number",
        "maximum": 99.99,
        "minimum": -99.99
      },
      "status": {
        "type": "string"
      },
      "birth": {
        "type": "string",
        "format": "date-time"
      },
      "year": {
        "type": "string",
        "pattern": "\\\\d{1,2}"
      },
      "initial": {
        "type": "string",
        "maxLength": 1
      },
      "avatar": {
        "type": "string"
      },
      "photo": {
        "type": "string"
      },
      "history": {
        "type": "string",
        "maxLength": 16777215
      },
      "gender": {
        "type": "string",
        "pattern": "^(M|F)(,(M|F))*$"
      },
      "intelligent": {
        "type": "string",
        "pattern": "^(N)(,(N))*$"
      },
      "shape": {
        "type": "string"
      },
      "pattern": {
        "type": "string"
      },
      "unique_id1": {
        "$comment": "primary key",
        "type": "string",
        "pattern": "^[a-f\\\\d]{8}-([a-f\\\\d]{4}-){3}[a-f\\\\d]{12}$"
      },
      "unique_id2": {
        "type": "string",
        "pattern": "^[a-f\\\\d]{8}-([a-f\\\\d]{4}-){3}[a-f\\\\d]{12}$"
      },
      "object": {
        "type": "string"
      }
    }
  },
  {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "JSON Schema for house table",
    "$id": "house",
    "title": "house",
    "type": "object",
    "required": [
      "id"
    ],
    "properties": {
      "id": {
        "$comment": "primary key",
        "type": "integer",
        "minimum": 1,
        "maximum": 2147483647
      },
      "pet_id": {
        "type": "integer",
        "minimum": -128,
        "maximum": 127
      },
      "coordx": {
        "type": "number",
        "maximum": 9999.99,
        "minimum": -9999.99
      },
      "coordy": {
        "type": "number"
      },
      "letter": {
        "type": "string",
        "maxLength": 1
      },
      "is_built": {
        "type": "boolean"
      },
      "is_apartment": {
        "type": "boolean"
      },
      "updated_at": {
        "type": "string"
      },
      "year": {
        "type": "string",
        "pattern": "\\\\d{1,4}"
      },
      "sequence": {
        "type": "string",
        "maxLength": 1,
        "default": "b'00101'"
      },
      "photo": {
        "type": "string"
      },
      "history": {
        "type": "string",
        "maxLength": 4294967295
      },
      "gender": {
        "type": "string",
        "enum": [
          "X"
        ]
      },
      "size": {
        "type": "string"
      },
      "where": {
        "type": "string"
      },
      "street": {
        "type": "string"
      },
      "neighbors": {
        "type": "string"
      },
      "city": {
        "type": "string"
      }
    }
  }
]
`;
